thirdparty/wasi/api.hpp

```diff
@@ -1751,6 +1752,7 @@ enum __wasi_sock_opt_so_t : uint32_t {
 
   __WASI_SOCK_OPT_SO_BINDTODEVICE = 14,
 
+  __WASI_SOCK_OPT_TCP_NODELAY = 15,
```

Comment:
The `__wasi_sock_opt_so_t` is for the socket options. Should we create a new struct `__wasi_sock_opt_tcp_t` for the TCP options?

Reply:
I guess so? Not sure how to adapt the set and get sock opt tho. Like this one for example https://github.com/WasmEdge/WasmEdge/blob/8c48ad0fb3feb1948cb0996f4afefae0957a459e/lib/host/wasi/inode-linux.cpp#L1214

Reply:
The function you said is internal and can be modified without breaking the external ABI.

This external function is used as ABI and interacts with other languages, such as Rust.
https://github.com/WasmEdge/WasmEdge/blob/master/lib/host/wasi/wasifunc.cpp#L2093-L2096